TYPO3.Flow/Classes/TYPO3/Flow/Http/Request.php

```diff
@@ -373,6 +417,136 @@ public function getContent($asResource = false)
     }
 
     /**
+     * Set a trusted proxy header.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.headers" instead.
+     *
+     * @param string $type One of the HEADER_* constants
+     * @param string $name The header name to trust or an empty string to not trust this header
+     * @return void
+     */
+    public static function setTrustedProxyHeader($type, $name)
+    {
+        if (!in_array($type, array(self::HEADER_CLIENT_IP, self::HEADER_HOST, self::HEADER_PORT, self::HEADER_PROTOCOL))) {
+            return;
+        }
+        self::$trustedProxiesSettings['headers'][$type] = strval($name);
+    }
+
+    /**
+     * Set the trusted proxies IP addresses.
+     * If the argument is an empty array, no proxies are trusted.
+     * If the argument is the string '*', all proxies are trusted.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.proxies" instead.
+     *
+     * @param array|string $trustedProxies
+     * @return void
+     */
+    public static function setTrustedProxies($trustedProxies)
+    {
```

Comment:
Currently it's actually a mixed type setting, since the '*' placeholder will denote a blanket whitelist. So either we then add a special method for that case (`trustAllProxies()` or sth.) or we change the configuration for that to ['*'] - but IMO that's odd, what if you have a list of IPs and then one entry '*'? That's ambiguous then, so I'm more leaning towards the special setter method